bin/main.rb

```diff
@@ -1,3 +1,97 @@
 #!/usr/bin/env ruby
 
-puts 'Hello World'
+class TicTacToe
+  def initialize
+    @turns = 0
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    assign_players
+    set_current_player
+    display_board
+    game_loop
+  end
+
+  def game_loop
+    game_on = true
+    while game_on
+      puts "it's #{@current_player}'s turn!"
+      choose_position
+      set_current_player
+      @turns += 1
+      if winner
+        draw
+        game_on = false
+      end
+    end
+  end
+
+  def winner
+    @turns == 3
+  end
+
+  def draw
+    puts 'Game Over!'
+    puts 'DRAW!'
+  end
+
+  def win
+    puts 'Game Over!'
+    puts "WIN! #{@current_player} won the game!"
+  end
+
+  def display_board
+    puts " #{@board[0]} | #{@board[1]} | #{@board[2]}"
+    puts " #{@board[3]} | #{@board[4]} | #{@board[5]}"
+    puts " #{@board[6]} | #{@board[7]} | #{@board[8]}"
+  end
+
+  def make_move(position)
+    @board[position] = 'X' if @current_player == @player_a
+    @board[position] = 'O' if @current_player == @player_b
+  end
+
+  def choose_position
+    puts "#{@current_player} Choose position to add your symbol to : 1-9"
+    choice = gets.chomp
+    choose_position if choice.empty? # restart the method if input is empty
+    choice = choice.to_i - 1
+    if valid_move?(choice)
+      make_move(choice)
+    else
+      puts 'Invalid move. Try again!'
+      choose_position
+    end
+  end
+
+  def set_current_player
+    @current_player = if @current_player == @player_a
+                        @player_b
+                      elsif @current_player == @player_b
+                        @player_a
+                      else
+                        [@player_a, @player_b].sample
+                      end
+  end
+
+  def assign_players
+    @player_a = input_names('X')
+    @player_b = input_names('O')
+  end
+
+  def input_names(player)
+    puts "Enter name for player #{player} "
+    input = gets.chomp
+    input_names(player) if input.empty? # restart the method if input is empty
+    input
+  end
+
+  def position_taken?(position)
+    @board[position] == 'X' || @board[position] == 'O'
+  end
+
+  def valid_move?(int)
+    !position_taken?(int) && int < @board.length
```

Comment:
- there is a bug here that lets the first player insert a letter or other character than a number between 1 and 9. Since the String class method .to_i will return 0 for a non-Number character, an error will not be thrown for the first turn. You can just add one more condition here (valid_move? method): ' ... && int > 0' .  
![Screenshot from 2021-03-24 00-04-36](https://user-images.githubusercontent.com/64894927/112204262-83a0d900-8c24-11eb-893c-36654a5b970d.png)